Backpack/src/main/java/net/skyscanner/backpack/starrating/BpkInteractiveStarRating.kt

```diff
@@ -76,6 +90,65 @@ open class BpkInteractiveStarRating @JvmOverloads constructor(
     return true
   }
 
+  override fun performAccessibilityAction(action: Int, arguments: Bundle?): Boolean {
+    if (super.performAccessibilityAction(action, arguments)) {
+      return true
+    }
+
+    when (action) {
+      AccessibilityNodeInfo.ACTION_SCROLL_BACKWARD -> {
+        rating -= 1f
+        updateStateDescription()
+        return true
+      }
+      AccessibilityNodeInfo.ACTION_SCROLL_FORWARD -> {
+        rating += 1f
+        updateStateDescription()
+        return true
+      }
+    }
+
+    return false
+  }
+
+  override fun onInitializeAccessibilityEvent(event: AccessibilityEvent) {
+    super.onInitializeAccessibilityEvent(event)
+    event.itemCount = maxRating
+    event.currentItemIndex = rating.toInt()
+    if (event.eventType == AccessibilityEvent.TYPE_VIEW_ACCESSIBILITY_FOCUSED) {
+      updateStateDescription()
+    }
+  }
+
+  override fun getAccessibilityClassName(): CharSequence {
```

Comment:
on pre-30 the accessibility class name somehow decides what controls are available